FRPs/proposal-38-optimal-mev.md

```diff
@@ -0,0 +1,36 @@
+---
+id: 38
+title: Approximating Optimal MEV extraction on Ethereum using Reinforcement Learning
+team: mevrbevr
+created: 2021-02-28
+paper: <paper the proposal is related to (1, 2, ...)>
+---
+
+# Approximating Optimal MEV extraction on Ethereum using Reinforcement Learning
+
+Here is a formal definition of what we mean by Optimal MEV: https://hackmd.io/EEbisqgATlW4eLKOIcsEYg?view (using HackMD becasue it renders LaTeX).
+
+In english: the optimal block producer is such that it will exract more than or equal MEV than any other block producer, for any possible chain state and mempool combination.
+
+I want to see what happens when you try to use reinforcement learning to approximate such a block producer. Here is a rough idea:
+- State space is S x B x M, where S it the current Ethereum state, B is a list of transaction currently proposed for the block, M is the txs remaning in mempool
```

Comment:
I'm a bit confused here--doesn't this state space leave out all MEV-extraction opportunities that were not yet spotted? What I'm getting at is that the agent would be navigating only proposed txs--either in the mempool or already "proposed for the block". How do we get it to go beyond what is "known" at search time?

Reply:
The new stuff would come out of P - the miner's personal transactions, defined as part of the action space. That P, which is the set of all valid transactions the miner can make at some state S, of course depends on the state at the block tip. But I am not sure we are on the same page here. Let me know if this clears anything up.

Reply:
Maybe the terminology is confusing, what I mean by state space would more accurately be expressed as the observation space I believe.

Reply:
Maybe a more informal view on all of this, which is the view that led me to think about RL in the first place:
You can imagine optimal MEV extraction as a game:

- You are given a deck of cards (the mempool)
- On each turn, either select a card from the deck and place it at the end of the row, or place your own "special" card (your own tx)
- The game ends when block is full (the cards fill up the block gas limit)
- Your score is how much MEV you got 

This "game" aspect led me to believe RL might yield results here, since RL was used successfully to play games before.

Reply:
Also in regards to "search", the space we are searching in, in this RL approach, is the space of strategies for this above mentioned game.

Reply:
Right, yeah, so my claim was that you seemed to me to be leaving the "novel" txs out of the strategy space.